subprojects/docs/src/docs/userguide/jvm/toolchains.adoc

```diff
@@ -289,6 +295,52 @@ Gradle will treat these directories as possible installations but will not desce
 org.gradle.java.installations.paths=/custom/path/jdk1.8,/shared/jre11
 ----
 
+[NOTE]
+====
+Custom toolchains are not prioritized over <<sec:auto_detection,auto-detected>> ones. If auto-detection is turned on, custom toolchains
+can still be specified but all they will do is to extend the set of detected JRE/JDK locations. Among these locations, one will be picked
+according to the <<sec:precedence,Toolchain Precedence Rules>>.
+====
+
+[[sec:precedence]]
+== Precedence
+
+When there are multiple local JRE/JDK installations matching the toolchain specification of a build, one will be picked according to the
+following rules:
+
+. JDK locations are preferred over JRE ones
+. certain vendors take precedence over others; their ordering (from the highest priority to lowest):
+.. ADOPTIUM
+.. ADOPTOPENJDK
+.. AMAZON
+.. APPLE
+.. AZUL
+.. BELLSOFT
+.. GRAAL_VM
+.. HEWLETT_PACKARD
+.. IBM
+.. IBM_SEMERU
+.. MICROSOFT
+.. ORACLE
+.. SAP
+.. everything else
+. higher major versions take precedence over lower ones
+. higher minor versions take precedence over lower ones
+. installation paths take precedence according to their lexicographic ordering (last resort criteria for deterministically deciding
+between installations of the same type, from the same vendor and with the same version)
+
+[NOTE]
+====
+The rules are applied in order, just like sorting over multiple fields. For example, if there is one JDK location
+detected and two JRE ones, then the JDK location will be picked, regardless of version numbers.
```

Comment:
This notes reads strangely. Especially that last comment about ignoring version numbers.

What matters is that once sorted, we evaluate them according to the predicate and the first one matching wins. Total ordering in that sense is irrelevant.

Reply:
Ok, that version number has been an unfortunate pick. But somehow we need to describe how exactly one will be picked from multiple matching installations. I'll rephrase.

Reply:
Haha! I will write an example!